File: lab1/wia1927_lab01.java
// Name         : Waseem Alkasbutrus
// ID           : 1001849127
// Lang Version : openjdk version "11.0.16" 2022-07-19
// OS           : Ubuntu 20.04.5 LTS (Not on a virtual machine)

// NOTE: Check README.md for answer to questions 1 and 2

import java.io.File;
import java.nio.file.Files;

class wia1927_lab01 {
    public static void main(String[] args) {
        long sum = sumFileSizes(new File("."));
        
        System.out.println(sum);
    }

    public static long sumFileSizes(File dir) {
        long size = 0;

        // get entries of dir
        String[] entries = dir.list();


        // for each object in the current directory
        for (String entryName : entries) {
            File entry = new File(dir.getAbsolutePath() + "/" + entryName);

            // check if it is a file or a directory
            if (entry.isFile() && entryName != "." && entryName != "..") {
                
                // Files.size can throw an exception, so use a try-catch block to catch any thrown exceptions 
                try {
                    // if it is a file, get its size
                    size += Files.size(entry.toPath());
                } catch (Exception e) {
                    System.out.println(e.getMessage());
                }

            } else if (entry.isDirectory() && entryName != "." && entryName != "..") {
                // if it is a dir, call this function again
                size += sumFileSizes(entry);
            }
        }
        
        return size;
    }
}

File: lab1/wia1927_lab01.py
# Name         : Waseem Alkasbutrus
# ID           : 1001849127
# Lang Version : Python 2.7.18
# OS           : Ubuntu 20.04.5 LTS (Not on a virtual machine)

# NOTE: Check README.md for answer to questions 1 and 2

import os

def sumFileSizes(dir):
    size = 0

    # get entries of dir
    entries = os.listdir(dir)

    # for each object in the current directory
    for entry in entries:

        # combine file/ directory name with current path to correctly read files/dirs in subdirectories
        fullEntry = os.path.join(dir, entry)

        # check if it is a file or a directory
        if os.path.isfile(fullEntry):
            # if it is a file, get its size
            size += os.path.getsize(fullEntry)
        elif os.path.isdir(fullEntry):
            # if it is a dir, call this function again
            size += sumFileSizes(fullEntry)
    
    return size

sum = sumFileSizes(os.getcwd())
print(sum)